### ISSUE TITLE
Parallel downloading/parsing

### ISSUE THREAD

**Issue by AMDmi3** [AMDmi3/repology#39]
Since repositories are independent, this should be easy to fetch and parse them in parallel. And it will really be useful for slow repositories such as pkgsrc (index generation) and Fedora (slow fetching)

**Comment by Cka3o4Huk** [comment]
Yes, it's sweet feature. import multiprocessing?

**Comment by AMDmi3** [comment]
Yep. I'll work on this after complete Fedora and pkgsrc support is implemented, since there's no much use for parallel processing now.

**Comment by AMDmi3** [comment]
This is done in a branch, but still needs testing and also some rework: fetching and parsing should be parallelized independently, because fetching is network-bound, and can run in parallel even on single CPU box. Especially this is useful when some repos fail to fetch, because the retry/timeout mechanism may add huge delays (timeout * retries * repos which usually fail in packs), while parsing is cpu-bound and it may be a good idea to limit number of processes for it, especially if it runs on the same box which servers HTTP.
